### PR TITLE
Job Worker Security

### PR DESCRIPTION
**Pull request by lucasbfernandes**
This is the final PR. It adds the security layers discussed on the design document.

Please follow the README instructions and use one of the following API keys to interact with the system:

* ADMIN - `qTMaYIfw8q3esZ6Dv2rQ`
* USER - `9EzGJOTcMHFMXphfvAuM`

Examples:

`./out/cli exec -t qTMaYIfw8q3esZ6Dv2rQ -s https://server-url:8080 sh -c "hello Im the admin user && ls wrongfile"`

`./out/cli exec -t 9EzGJOTcMHFMXphfvAuM -s https://server-url:8080 sh -c "hello Im the user && ls wrongfile"`

### CODE REVIEW

File: server/internal/controllers/controllers.go [hunk 1]
```diff
@@ -68,17 +93,101 @@ func (s *Server) createLogsDir() error {
...
+		if apiToken, ok := claims["apiToken"]; !ok || apiToken == "" {
+			c.AbortWithStatus(http.StatusUnauthorized)
+			return
+		}
+
+		c.Set("apiToken", claims["apiToken"])
+	}
+}
+
+func (s *Server) UserExistenceGuard() gin.HandlerFunc {
```
Comment (Joerger) [inline server/internal/controllers/controllers.go]: nit: I would just name this `Authenticate`

File: server/internal/controllers/controllers.go [hunk 2]
```diff
@@ -68,17 +93,101 @@ func (s *Server) createLogsDir() error {
...
+		if err != nil {
+			c.AbortWithStatus(http.StatusNotFound)
+			return
+		}
+
+		c.Set("user", user)
+	}
+}
+
+func (s *Server) UserJobGuard() gin.HandlerFunc {
```
Comment (Joerger) [inline server/internal/controllers/controllers.go]: nit: this can just be `Authorize`

**Review by Joerger — APPROVED** [review]

File: server/internal/controllers/controllers.go (hunk 1 above)
Comment (lucasbfernandes) [reply server/internal/controllers/controllers.go]: Done here https://github.com/lucasbfernandes/job-worker/commit/bcb3ecb0f5428a2805aefb881bb01573679cd70c

File: server/internal/controllers/controllers.go (hunk 2 above)
Comment (lucasbfernandes) [reply server/internal/controllers/controllers.go]: Done here https://github.com/lucasbfernandes/job-worker/commit/bcb3ecb0f5428a2805aefb881bb01573679cd70c

**Pull request merged**